File: GameHive/Constants/DifficultyLevelEnum.cs
/*************************************************************************************
 * 文 件 名:   DifficultyLevelEnum.cs
 * 描    述:   游戏难度等级
 * 版    本：  V3.0 引入DRL、MinMax-MCTS混合算法
 * 创 建 者：  Cassifa
 * 创建时间：  2025/5/3 15:19
*************************************************************************************/
namespace GameHive.Constants.DifficultyLevelEnum {
    public enum DifficultyLevel {
        LEVEL_1 = 1,
        LEVEL_2 = 2,
        LEVEL_3 = 3,
        LEVEL_4 = 4,
        LEVEL_5 = 5
    }

    public static class DifficultyLevelExtensions {
        // 获取中文难度名称
        public static string GetChineseName(this DifficultyLevel level) => level switch {
            DifficultyLevel.LEVEL_1 => "一级",
            DifficultyLevel.LEVEL_2 => "二级",
            DifficultyLevel.LEVEL_3 => "三级",
            DifficultyLevel.LEVEL_4 => "四级",
            DifficultyLevel.LEVEL_5 => "五级",
            _ => "未知"
        };

        // 获取英文难度名称
        public static string GetEnglishName(this DifficultyLevel level) => level switch {
            DifficultyLevel.LEVEL_1 => "Level 1",
            DifficultyLevel.LEVEL_2 => "Level 2",
            DifficultyLevel.LEVEL_3 => "Level 3",
            DifficultyLevel.LEVEL_4 => "Level 4",
            DifficultyLevel.LEVEL_5 => "Level 5",
            _ => "Unknown"
        };

        public static List<DifficultyLevel> GetLevelRange(int maxLevel) {
            if (maxLevel < 1 || maxLevel > 5)
                throw new ArgumentOutOfRangeException("难度等级必须是1-5");

            return Enumerable.Range(1, maxLevel)
                .Select(x => (DifficultyLevel)x)
                .ToList();
        }
        public static bool IsGreaterThan(this DifficultyLevel current, DifficultyLevel other)
            => (int)current > (int)other;

        public static bool IsLessThan(this DifficultyLevel current, DifficultyLevel other)
            => (int)current < (int)other;

        public static bool IsEqualTo(this DifficultyLevel current, DifficultyLevel other)
            => current == other;
    }
}

File: Backend/gamehive-admin/src/main/java/com/gamehive/constants/DifficultyLevelEnum.java
package com.gamehive.constants;

import java.util.ArrayList;
import java.util.List;

/**
 * 游戏难度等级枚举
 * @author Claude
 */
public enum DifficultyLevelEnum {
    LEVEL_1(1, "一级", "Level 1"),
    LEVEL_2(2, "二级", "Level 2"),
    LEVEL_3(3, "三级", "Level 3"),
    LEVEL_4(4, "四级", "Level 4"),
    LEVEL_5(5, "五级", "Level 5");

    private final int level;
    private final String chineseName;
    private final String englishName;

    DifficultyLevelEnum(int level, String chineseName, String englishName) {
        this.level = level;
        this.chineseName = chineseName;
        this.englishName = englishName;
    }

    public int getLevel() {
        return level;
    }

    public String getChineseName() {
        return chineseName;
    }

    public String getEnglishName() {
        return englishName;
    }

    public static List<DifficultyLevelEnum> getLevelRange(int maxLevel) {
        if (maxLevel < 1 || maxLevel > 5) {
            throw new IllegalArgumentException("难度等级必须是1-5");
        }

        List<DifficultyLevelEnum> levels = new ArrayList<>();
        for (DifficultyLevelEnum level : values()) {
            if (level.getLevel() <= maxLevel) {
                levels.add(level);
            }
        }
        return levels;
    }

    public boolean isGreaterThan(DifficultyLevelEnum other) {
        return this.level > other.level;
    }

    public boolean isLessThan(DifficultyLevelEnum other) {
        return this.level < other.level;
    }

    public boolean isEqualTo(DifficultyLevelEnum other) {
        return this.level == other.level;
    }

    public static DifficultyLevelEnum fromLevel(int level) {
        if (level < 1 || level > 5) {
            throw new IllegalArgumentException("难度等级必须是1-5");
        }
        
        for (DifficultyLevelEnum value : values()) {
            if (value.getLevel() == level) {
                return value;
            }
        }
        throw new IllegalArgumentException("未知难度等级: " + level);
    }
} 